src/lib/y2network/interface_config_builders/wireless.rb

```diff
@@ -0,0 +1,98 @@
+require "yast"
+require "forwardable"
+require "y2network/config"
+require "y2network/interface_config_builder"
+
+module Y2Network
+  module InterfaceConfigBuilders
+    # Builder for wireless configuration. Many methods delegated to ConnectionConfig::Wireless
+    class Wireless < InterfaceConfigBuilder
+      extend Forwardable
+      include Yast::Logger
+
+      def initialize(config: nil)
+        super(type: InterfaceType::WIRELESS, config: config)
+      end
+
+      def mode
+        select_backend(
+          @config["WIRELESS_MODE"],
+          @connection_config.mode
+        )
+      end
+
+      def mode=(mode)
+        @config["WIRELESS_MODE"] = mode
+        @connection_config.mode = mode
+      end
+
+      def essid
+        select_backend(
+          @config["WIRELESS_ESSID"],
+          @connection_config.essid
+        )
+      end
+
+      def essid=(value)
+        @config["WIRELESS_ESSID"] = value
+        @connection_config.essid = value
+      end
+
+      def auth_modes
+        Yast::LanItems.wl_auth_modes
```

Comment:
Something that we will need to add to our model. But let's do it when we drop `LanItems`.

Reply:
I already added note to trello. It is not so hard. It is just getting that modes from hardware.